fact-ebpf/src/bpf/bound_path.h

```diff
@@ -35,6 +39,14 @@ __always_inline static struct bound_path_t* path_read(struct path* path) {
   return bound_path;
 }
 
+__always_inline static struct bound_path_t* path_read(struct path* path) {
+  return _path_read(path, true);
+}
+
+__always_inline static struct bound_path_t* path_read_alternate(struct path* path) {
+  return _path_read(path, false);
+}
+
```

Comment:
Since we have access to the boolean, can we just have `path_read`? Or does the verifier pull a fast one on us?


```suggestion
__always_inline static struct bound_path_t* path_read(struct path* path) {
  return _path_read(path, path_unlink_supports_bpf_d_path);
}
```

Reply:
The problem with this is that `path_read` is used from both the `path_unlink` and `file_open` LSM hooks, AFAICT, the `file_open` hook can always use `bpf_d_path` so it would be counterproductive to remove it when `path_unlink` doesn't support it.

I also didn't want to change the signature for `path_read`, since it is more intuitive to just see something like `struct bound_path_t* path = path_read(&file->f_path);` than `struct bound_path_t* path = path_read(&file->f_path, true);` (we shouldn't have to go look at the signature to understand what `true` is).

In any case, I did end up making this change in #134, so maybe I just pull it down and make the signature:
```c
__always_inline static struct bound_path_t* path_read(struct path* path, bool use_bpf_d_path);
```

WDYT?

Reply:
hmm yeah good point. I agree that the signature is much nicer without the extra field; so perhaps a rename of `path_read_alternate` to `path_read_no_d_path` to make the meaning super clear?